implementations/micrometer-registry-signalfx/src/test/java/io/micrometer/signalfx/SignalFxMeterRegistryTest.java

```diff
@@ -457,13 +456,13 @@ void shouldNotExportCumulativeHistogramDataByDefault_Timer() {
         mockClock.add(config.step().minus(Duration.ofMillis(1)));
 
         assertThat(getDataPoints(registry, mockClock.wallTime())).hasSize(8)
-                .has(gaugePoint("my.timer.avg", 0.2525), atIndex(0)).has(counterPoint("my.timer.count", 2), atIndex(1))
+                .has(gaugePoint("my.timer.avg", 2.525), atIndex(0)).has(counterPoint("my.timer.count", 2), atIndex(1))
```

Comment:
My review comment about investigating the cause of the failures meant understanding exactly how the behavior changed and make sure the change and test made sense. This appears to change the values so the test passes, but I don't think it's demonstrating what the test intended. We need to be clear about what changed and if that's acceptable. I'll write up a summary in another comment.

When investigating this, I found a bug in the assertion logic for `hasValue` which caused it to always return true for double values that would have an int value of 0 (like 0.2525). I'll fix that in a separate issue and rebase this pull request on it.

Reply:
As you pointed out `hasValue` had a bug all the time and the tests would have failed if it is right. And the other issue is with the TimeWindowFixedBoundaryHistogram and buffer length. That is the whole reason SignalFx Registry went for a Delta Histogram Implementation.

Reply:
I've changed the tests back to assert what they did before as I think that's what the intention of the tests was. I've had to change the test to record later than the beginning of the step to avoid the histogram counts being rotated out when checking the published data. That whole problem is a preexisting issue that was worked around in the tests by checking the published data at a specific millisecond before the histogram would rotate but after the step-based values (count/sum/avg) would return the desired new values. We can consider if the behavior of histogram rotation is best or not separately from this pull request.

Reply:
Thanks.